### PR TITLE
change embedded k8s version to v1.23.7

### PR DESCRIPTION
**Pull request by ibrokethecloud**
The PR introduces the following changes:

* bumps up k8s and associated components to v1.23.7. This is inline with harvester v1.1.0 roadmap
* bumps to go1.18
* adds clean for ingress objects to convert from ingress.extensions to ingress.networking as the former is no longer available in v1.23 but we need to be able to still process older support bundles
* change eventsTTL in embedded apiserver, to 90 days. This fixes the issue where events from older bundles get GC'd right after the import.

### CODE REVIEW

**Comment by ibrokethecloud** [comment]
Related to: https://github.com/harvester/harvester/issues/3061

File: scripts/test [hunk 1]
```diff
@@ -10,4 +10,4 @@ echo "running unit tests"
 go test -p 1 -v ./pkg/simulator/...
 
 echo "running integration tests"
-go test -v ./tests/integration
\ No newline at end of file
+go test -v -timeout=20m ./tests/integration
```
[the remark this answers is not in the document]
Comment (ibrokethecloud) [reply scripts/test:13]: there is a full support bundle in there and loading resources one at a time can take more than 10 mins.

File: pkg/simulator/objects/cleanup.go [hunk 2]
```diff
@@ -259,3 +259,11 @@ func cleanupIngress(obj *unstructured.Unstructured) error {
 
 	return nil
 }
+
+// cleanupCRD will cleanup CRDs by removing .spec.conversion
+// this is not needed as the object conversion has already been performed in the
+// source cluster
+func cleanupCRD(obj *unstructured.Unstructured) error {
```
[the remark this answers is not in the document]
Comment (ibrokethecloud) [reply pkg/simulator/objects/cleanup.go:266]: removing the field would set it to `nil` and that allows objects to be processed.

I will clean up the method name

**Review by bk201 — APPROVED** [review]
LGTM!

**Review by guangbochen — APPROVED** [review]
LGTM, thanks

File: scripts/test (hunk 1 above)
Comment (Vicente-Cheng) [inline scripts/test:13]: I thought the new test would be good. But could we add some variable to skip the test SB loading to increase the local build?

**Review by Vicente-Cheng — COMMENTED** [review]
overall LGTM, just one nit problem.
Already test it with v1.1.0 SB, works well!

**Review by Vicente-Cheng — APPROVED** [review]
BTW, you could merge it anytime. My opinion (on the above) is not the blocker for this PR.

That PR is LGTM!

File: scripts/test (hunk 1 above)
Comment (ibrokethecloud) [reply scripts/test:13]: i agree we could skip the integration tests via a variable and only use the integration tests to run for tagged builds.

**Pull request merged**